sklearn/utils/estimator_checks.py

```diff
@@ -832,25 +842,45 @@ def check_sample_weights_invariance(name, estimator_orig):
         X = np.array([[1, 3], [1, 3], [1, 3], [1, 3],
                       [2, 1], [2, 1], [2, 1], [2, 1],
                       [3, 3], [3, 3], [3, 3], [3, 3],
-                      [4, 1], [4, 1], [4, 1], [4, 1]], dtype=np.dtype('float'))
+                      [4, 1], [4, 1], [4, 1], [4, 1]], dtype=np.float64)
         y = np.array([1, 1, 1, 1, 2, 2, 2, 2,
-                      1, 1, 1, 1, 2, 2, 2, 2], dtype=np.dtype('int'))
+                      1, 1, 1, 1, 2, 2, 2, 2], dtype=np.int)
+
+        if kind == 'ones':
+            X2 = X
+            y2 = y
+            sw2 = None
```

Comment:
Would it be more natural to set this to ones, and set the SW of estimator1 to None instead?

Reply:
Indeed, fixed.